app/services/forest_liana/search_query_builder.rb

```diff
@@ -79,13 +79,29 @@ def search_param
     def filter_param
       if @params[:filterType] && @params[:filter]
         conditions = []
+
         @params[:filter].each do |field, values|
-          next if association?(field)
+          # ActsAsTaggable
+          if acts_as_taggable?(field)
+            tagged_records = @records.tagged_with(values.tr('*', ''))
+
+            if @params[:filterType] == 'and'
+              @records = tagged_records
+            elsif @params[:filterType] == 'or'
```

Comment:
If you can do that for the filters, why not doing it this way (double request) for the search?